open-apparel-registry/open-apparel-registry
# Add row_index to FacilityListItem

## Overview

Adds a `row_index` field to `FacilityListItem` to make uploading to AWS Batch Array Jobs easier.

Also cleans up some lint.

Connects #160 

## Demo

```sql
SELECT id, row_index, status, processing_started_at, processing_completed_at, processing_results, facility_id, facility_list_id
FROM api_facilitylistitem
LIMIT 5;

 id | row_index |  status  | processing_started_at | processing_completed_at | processing_results | facility_id | facility_list_id
----+-----------+----------+-----------------------+-------------------------+--------------------+-------------+------------------
  1 |         0 | UPLOADED |                       |                         | {}                 |             |                1
  2 |         1 | UPLOADED |                       |                         | {}                 |             |                1
  3 |         2 | UPLOADED |                       |                         | {}                 |             |                1
  4 |         3 | UPLOADED |                       |                         | {}                 |             |                1
  5 |         4 | UPLOADED |                       |                         | {}                 |             |                1
(5 rows)
```

```
\d api_facilitylistitem;
                                             Table "public.api_facilitylistitem"
         Column          |           Type           | Collation | Nullable |                     Default
-------------------------+--------------------------+-----------+----------+--------------------------------------------------
 id                      | integer                  |           | not null | nextval('api_facilitylistitem_id_seq'::regclass)
 row_index               | integer                  |           | not null |
 raw_data                | text                     |           | not null |
 status                  | character varying(200)   |           | not null |
 processing_started_at   | timestamp with time zone |           |          |
 processing_completed_at | timestamp with time zone |           |          |
 processing_results      | jsonb                    |           | not null |
 name                    | character varying(200)   |           | not null |
 address                 | character varying(200)   |           | not null |
 country_code            | character varying(2)     |           | not null |
 geocoded_point          | geometry(Point,4326)     |           |          |
 geocoded_address        | character varying(200)   |           |          |
 created_at              | timestamp with time zone |           | not null |
 updated_at              | timestamp with time zone |           | not null |
 facility_id             | character varying(32)    |           |          |
 facility_list_id        | integer                  |           | not null |
Indexes:
    "api_facilitylistitem_pkey" PRIMARY KEY, btree (id)
    "api_facilit_facilit_196927_idx" btree (facility_list_id, row_index)
    "api_facilitylistitem_facility_id_7c94a3a3" btree (facility_id)
    "api_facilitylistitem_facility_id_7c94a3a3_like" btree (facility_id varchar_pattern_ops)
    "api_facilitylistitem_facility_list_id_ad3f851d" btree (facility_list_id)
    "api_facilitylistitem_geocoded_point_id" gist (geocoded_point)
Foreign-key constraints:
    "api_facilitylistitem_facility_id_7c94a3a3_fk_api_facility_id" FOREIGN KEY (facility_id) REFERENCES api_facility(id) DEFERRABLE INITIALLY DEFERRED
    "api_facilitylistitem_facility_list_id_ad3f851d_fk_api_facil" FOREIGN KEY (facility_list_id) REFERENCES api_facilitylist(id) DEFERRABLE INITIALLY DEFERRED
Referenced by:
    TABLE "api_facility" CONSTRAINT "api_facility_created_from_id_3971db46_fk_api_facil" FOREIGN KEY (created_from_id) REFERENCES api_facilitylistitem(id) DEFERRABLE INITIALLY DEFERRED
    TABLE "api_facilitymatch" CONSTRAINT "api_facilitymatch_facility_list_item_i_ac429b38_fk_api_facil" FOREIGN KEY (facility_list_item_id) REFERENCES api_facilitylistitem(id) DEFERRABLE INITIALLY DEFERRED
```

## Testing Instructions

* Check out this branch and clear your schema and run migrations
* Upload a CSV and inspect the `api_facilitylistitem` table
* Ensure `row_index` is correctly populated
